src/main/java/Erika.java

```diff
@@ -0,0 +1,162 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Erika {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static Scanner in = new Scanner(System.in);
+    private static int markIndex;
+    
+    public static void main(String[] args) {
+        printWelcomeMessage();
+        while (true) {
+            String line = collectUserInput();
+            if (line.equals("bye")) {
+                printGoodbyeMessage();
+                break;
+            } else if (line.equals("list")) {
+                printList();
+            } else if (line.contains("mark ")) {
+                handleMark(line);
+            } else if (line.contains("todo ")) {
+                addTodo(line);
+            } else if (line.contains("deadline ")) {
+                addDeadline(line);
+            } else if (line.contains("event ")) {
+                addEvent(line);
```

Comment:
Polymorphism could be used here to reduce the need for handling different cases.

Reply:
Sorry I don't quite understand how said polymorphism will work since I need to deliberately create a new object of the respective subclass (Todo, Deadline, Event) when adding to the task array. Therefore I am not sure how I can leverage polymorphism to call a single (overriden) method for different cases. Thanks!